cmake/External/aotriton.cmake

```diff
@@ -37,17 +56,27 @@ if(NOT __AOTRITON_INCLUDED)
     add_dependencies(__caffe2_aotriton aotriton_external)
     message(STATUS "Using AOTriton compiled from source directory ${__AOTRITON_EXTERN_PREFIX}")
   else()
-    file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/.ci/docker/aotriton_version.txt" __AOTRITON_CI_INFO)
-    list(GET __AOTRITON_CI_INFO 0 __AOTRITON_VER)
-    list(GET __AOTRITON_CI_INFO 1 __AOTRITON_MANYLINUX)
-    list(GET __AOTRITON_CI_INFO 2 __AOTRITON_ROCM)
-    list(GET __AOTRITON_CI_INFO 3 __AOTRITON_COMMIT)
-    list(GET __AOTRITON_CI_INFO 4 __AOTRITON_SHA256)
+    list(GET __AOTRITON_ROCM_LIST 0 __AOTRITON_ROCM_LOW_STR)
+    list(GET __AOTRITON_ROCM_LIST -1 __AOTRITON_ROCM_HIGH_STR)
+    # len("rocm") == 4
+    string(SUBSTRING ${__AOTRITON_ROCM_LOW_STR} 4 -1 __AOTRITON_ROCM_LOW)
+    string(SUBSTRING ${__AOTRITON_ROCM_HIGH_STR} 4 -1 __AOTRITON_ROCM_HIGH)
+    set(__AOTRITON_SYSTEM_ROCM "${ROCM_VERSION_DEV_MAJOR}.${ROCM_VERSION_DEV_MINOR}")
+    if(__AOTRITON_SYSTEM_ROCM VERSION_LESS __AOTRITON_ROCM_LOW)
+      set(__AOTRITON_ROCM ${__AOTRITON_ROCM_LOW})
+    elseif(__AOTRITON_SYSTEM_ROCM VERSION_GREATER __AOTRITON_ROCM_HIGH)
+      set(__AOTRITON_ROCM ${__AOTRITON_ROCM_HIGH})
+    else()
+      set(__AOTRITON_ROCM ${__AOTRITON_SYSTEM_ROCM})
```

Comment:
Does this logic not require publishing aotriton tarballs for each ROCm version, even if there's no ABI breakage from one ROCm version to the next?

Reply:
Yes, but it won't happen unless AOTriton decides to skip a certain version of ROCM (which is unlikely)

Reply:
Done